src/libstd/panicking.rs

```diff
@@ -243,8 +247,8 @@ fn default_hook(info: &PanicInfo) {
 
             static FIRST_PANIC: AtomicBool = AtomicBool::new(true);
 
-            if log_backtrace {
-                let _ = backtrace::write(err);
+            if let Some(format) = log_backtrace {
+                let _ = backtrace::print(err, format);
             } else if FIRST_PANIC.compare_and_swap(true, false, Ordering::SeqCst) {
                 let _ = writeln!(err, "note: Run with `RUST_BACKTRACE=1` for a backtrace.");
```

Comment:
The help message below should be updated to say something like
    note: Run with `RUST_BACKTRACE=full` or `RUST_BACKTRACE=short` for a backtrace.